conans/client/loader.py

```diff
@@ -70,8 +71,11 @@ def __init__(self, runner, settings, profile):
         @param cached_env_values: EnvValues object
         """
         self._runner = runner
+
+        assert isinstance(settings, Settings)
         settings.values = profile.settings_values
-        assert settings is None or isinstance(settings, Settings)
+        preprocess(settings, output)
```

Comment:
Why this responsability inside the "Loader"? I think it should be independent, and the settings provided to the loader should be already complete and defined.